src/libraries/System.Private.CoreLib/src/System/Byte.cs

```diff
@@ -272,5 +279,704 @@ object IConvertible.ToType(Type type, IFormatProvider? provider)
+        // static checked byte IMultiplyOperators<byte, byte, byte>.operator *(byte left, byte right)
+        //     => checked((byte)(left * right));
+
+        //
+        // INumber
+        //
+
+        [RequiresPreviewFeatures]
+        static byte INumber<byte>.One => 1;
+
+        [RequiresPreviewFeatures]
+        static byte INumber<byte>.Zero => 0;
+
+        [RequiresPreviewFeatures]
+        static byte INumber<byte>.Abs(byte value)
+            => value;
+
+        [RequiresPreviewFeatures]
+        static byte INumber<byte>.Clamp(byte value, byte min, byte max)
+            => Math.Clamp(value, min, max);
+
+        [RequiresPreviewFeatures]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static byte INumber<byte>.Create<TOther>(TOther value)
+        {
+            if (typeof(TOther) == typeof(byte))
+            {
+                return (byte)(object)value;
+            }
+            else if (typeof(TOther) == typeof(char))
+            {
+                return checked((byte)(char)(object)value);
+            }
+            else if (typeof(TOther) == typeof(decimal))
+            {
+                return checked((byte)(decimal)(object)value);
+            }
+            else if (typeof(TOther) == typeof(double))
+            {
+                return checked((byte)(double)(object)value);
+            }
+            else if (typeof(TOther) == typeof(short))
+            {
+                return checked((byte)(short)(object)value);
+            }
+            else if (typeof(TOther) == typeof(int))
+            {
+                return checked((byte)(int)(object)value);
+            }
+            else if (typeof(TOther) == typeof(long))
+            {
+                return checked((byte)(long)(object)value);
+            }
+            else if (typeof(TOther) == typeof(nint))
+            {
+                return checked((byte)(nint)(object)value);
+            }
+            else if (typeof(TOther) == typeof(sbyte))
+            {
+                return checked((byte)(sbyte)(object)value);
+            }
+            else if (typeof(TOther) == typeof(float))
+            {
+                return checked((byte)(float)(object)value);
+            }
+            else if (typeof(TOther) == typeof(ushort))
+            {
+                return checked((byte)(ushort)(object)value);
+            }
+            else if (typeof(TOther) == typeof(uint))
+            {
+                return checked((byte)(uint)(object)value);
+            }
+            else if (typeof(TOther) == typeof(ulong))
+            {
+                return checked((byte)(ulong)(object)value);
+            }
+            else if (typeof(TOther) == typeof(nuint))
+            {
+                return checked((byte)(nuint)(object)value);
+            }
+            else
+            {
+                ThrowHelper.ThrowNotSupportedException();
+                return default;
+            }
+        }
```

Comment:
In the Create\<TOther\>() methods implementation,
why the type check is based on `if/else if`, `typeof(TOther) == typeof(int)` and cast `return checked((byte)(int)(object)value);`
Instead of:
```csharp
static byte INumber<byte>.Create<TOther>(TOther value)
=> value switch
  {
    byte b => b,
    int i => checked((byte)i),
    ...
  }
```

Reply:
when the rhs is value type, `typeof(TOther) == typeof(int)` and `(int)(object)value` will be replaced by constant by JIT. The pattern generated by switch expression is not recognized by JIT and will be boxed.

Reply:
Ok, I was curious and I tried both style in [Sharplab.io](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzmAPgAQEwAYCwAUDgIxGa4AEAygBYQBOADgDITAB0ASgK4wJQwAUwDcRciQCcACgDCDIRARDqAdygIAxrWnTgAT2UBKAKxGjY4lLkKlK9Vp0mW5y5mvzFygCr6mQgHkAM10DYzMLCRlPO19/YOlnV3FCAG0AKQ0AcSEYIQYoTWkEPyEAexCwoXM4SkyEHLyCopL/CukAGzKYAHMauuzc/MLi0vaAEzKeYA7qowBdCQA2SlgEShjlNQ1tAB5vAIRafIA+aQOj/MoANwgOnmqiSmfKAF4Tm7uHyhQHbSeXgBvAEvUEvKqUYBvD7AOAgsGgrq9SgdaGUaRrIwdOGEBEIgD6aKODDKqkoeTJAFEAB6aIRMATdaRGeEAX0sRAygyaI1a5UqhjmtXqjWGLTGISRfSMwu5YtGbRCk2ms3MiysKzWG1sPlKwX2h2ODDOFyNn3uj0IwNxLygQXRfPapvyRjer0ojoFxhZNue1rxlEwAHZ0VUjNIysAAFZCTQIIy3C2WUGs+FCDooISre0K/nnQ0ut0eiWdbrS+H+vHB9GY0u9cORmNxhNfURstMZoQV+Gg4mk8lCKm0+mMmDM5MvVOEVlAA), the JIT generates exactly the same `switch` and `typeof` methods.
In the IL, both versions will box the values.

Reply:
The IL generated is very different and the JIT has a longstanding and good history with the `if (typeof(TOther) == typeof(...))` pattern. It's only fairly recently started to recognize the switch pattern and there may still be cases where it doesn't do the right thing, so I went with the "safer" route.

Reply:
Ok, Thanks for the answers.
I was just curious for why things have been made that way.